AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/handler/PostHandler.java

```diff
@@ -42,4 +72,74 @@ protected String getUrlSuffix(Element element) {
 		Post annotation = element.getAnnotation(Post.class);
 		return annotation.value();
 	}
+
+	@Override
+	protected JExpression getRequestEntity(ExecutableElement element, RestHolder holder, JBlock methodBody, SortedMap<String, JVar> params) {
+		JVar httpHeaders = restAnnotationHelper.declareHttpHeaders(element, holder, methodBody);
+		JVar entitySentToServer = restAnnotationHelper.getEntitySentToServer(element, params);
+
+		if (entitySentToServer == null) {
+			Map<String, String> postParameters = restAnnotationHelper.extractPostParameters(element);
+
+			if (postParameters != null) {
+				JClass hashMapClass = getJClass(RestSpringClasses.LINKED_MULTI_VALUE_MAP).narrow(String.class, Object.class);
+				entitySentToServer = methodBody.decl(hashMapClass, "postParameters", JExpr._new(hashMapClass));
+
+				for (Map.Entry<String, String> postParameter : postParameters.entrySet()) {
+					methodBody.add(entitySentToServer.invoke("add").arg(JExpr.lit(postParameter.getKey())).arg(params.get(postParameter.getValue())));
+				}
+			}
+		}
+
+		return restAnnotationHelper.declareHttpEntity(methodBody, entitySentToServer, httpHeaders);
+	}
+
+	private abstract class AbstractPostParamHandler extends BaseAnnotationHandler<GeneratedClassHolder> {
+
+		AbstractPostParamHandler(Class<?> targetClass, AndroidAnnotationsEnvironment environment) {
+			super(targetClass, environment);
+		}
+
+		@Override
+		public void process(Element element, GeneratedClassHolder holder) throws Exception {
```

Comment:
Same here. `process` is called after `validate` so it seems more logical to me to put it after `validate`.